### PR TITLE
make expressions generic over columns

### PR DESCRIPTION
**Pull request by fabrizio-m**
Adding a generic to `Expr` allowing to be generic over the columns, also adding an implementation for `Column`.

### CODE REVIEW

**Comment by fabrizio-m** [comment]
!ci-build-me

File: kimchi/src/circuits/expr.rs [hunk 1]
```diff
@@ -468,20 +546,20 @@ impl FeatureFlag {
 /// the corresponding combination of the polynomials corresponding to
 /// the above variables should vanish on the PLONK domain.
 #[derive(Clone, Debug, PartialEq)]
-pub enum Expr<C> {
+pub enum Expr<C, Col: ColTrait = Column> {
```
Comment (querolita) [inline kimchi/src/circuits/expr.rs:549]: I am a bit confused aboout this notation. 
1. What exactly was `C`?
2. Is `Column` the kind of "columns" we have in Berkeley Kimchi, and if we had a different configuration with a different system we would have to re-write a new `Expr` whose `ColTrait` was `ColumnNew` for example?

**Review by querolita — APPROVED** [review]
Nice looking so far :) 

I left one doubt regarding different column configurations, please reply. 

I made some changes to the witness interface to work generically over any number of columns [here](https://github.com/o1-labs/proof-systems/edit/zkvm/keccak/layout/kimchi/src/circuits/witness/mod.rs?pr=%2Fo1-labs%2Fproof-systems%2Fpull%2F1248) 

It is not necessary using that interface to write the witness generation code for a gate. But in any case, once both PRs (this and that one) are merged, I will be happy to help unify the notation so it matches this one's better, if you think it could make sense.

**Comment by mrmr1993** [comment]
There's an equivalent version of this already merged to `master`. Closing.

**Pull request closed without merging**